python/python/lancedb/table.py

```diff
@@ -1012,6 +1012,24 @@ def drop_columns(self, columns: Iterable[str]):
             The names of the columns to drop.
         """
 
+    @abstractmethod
+    def checkout(self):
+        """Checkout a version of the table. This is an in-place operation.
+
+        This allows viewing previous versions of the table.
+
+        Calling this method will set the table into time-travel mode. If you
+        wish to return to standard mode, call `checkout_latest`.
```

Comment:
I might be nice to link to the spot where time-travel mode is defined but not mandatory